Mathlib/Algebra/BigOperators/Balance.lean

```diff
@@ -0,0 +1,59 @@
+/-
+Copyright (c) 2024 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import Mathlib.Algebra.BigOperators.Expect
+
+/-!
+# Balancing a function
+
+This file defines the balancing of a function `f`, defined as `f` minus its average.
+
+This is the unique function `g` such that `f a - f b = g a - g b` for all `a` and `b`, and
+`∑ a, g a = 0`. This is particularly useful in Fourier analysis as `f` and `g` then have the same
+Fourier transform, except in the `0`-th frequency where the Fourier transform of `g` vanishes.
+-/
+
+open Function
+open Fintype (card)
+open scoped BigOperators Pointwise NNRat
+
+variable {ι F G K : Type*}
+
+namespace Finset
```

Comment:
Why do you put this into the `Finset` namespace while there is no finset below? `Fintype`, maybe?

Reply:
Or `Function` to enable dot notation?

Reply:
That was to match `Finset.expect`, but you are right that `Fintype` is strictly better. I think it shouldn't be in the `Function` namespace however, since there is an `integral` version that could also claim that name.